simnode/generator.py

```diff
@@ -156,44 +156,47 @@ def add_obstacles_to_scenario(self, scenario: Scenario) -> None:
                 mesh = ProceduralBump(pos, rot, obstacle.width, obstacle.length, height, obstacle.upper_length,
                                       obstacle.upper_width)
             elif obstacle_type is Stopsign:
+                rot = (0, 0, -(obstacle.z_rot - 90))
                 id_number = randrange(1000)
                 name_sign = "stopsign" + str(id_number)
-                stopsign = StaticObject(pos=(obstacle.x, obstacle.y, 0), rot=rot, name=name_sign, scale=(3, 3, 3),
+                stopsign = StaticObject(pos=(obstacle.x, obstacle.y, 0), rot=rot, name=name_sign, scale=(1.9, 1.9, 1.9),
                                         shape='/levels/drivebuild/art/objects/stopsign.dae')
                 scenario.add_object(stopsign)
             elif obstacle_type is TrafficLightSingle:
                 id_number = randrange(1000)
                 name_light = "trafficlight" + str(id_number)
                 name_pole = "pole" + str(id_number)
-                traffic_light = StaticObject(name=name_light, pos=(obstacle.x, obstacle.y, 5.32), rot=rot,
+                rot=(0, 0, -(obstacle.z_rot) - 90)
+                traffic_light = StaticObject(name=name_light, pos=(obstacle.x, obstacle.y, 4.62), rot=rot,
                                              scale=(1, 1, 1),
                                              shape='/levels/drivebuild/art/objects/trafficlight1a.dae')
                 scenario.add_object(traffic_light)
-                pole = StaticObject(name=name_pole, pos=(obstacle.x, obstacle.y, 0), rot=rot, scale=(1, 1, 1.3),
+                pole = StaticObject(name=name_pole, pos=(obstacle.x, obstacle.y, 0), rot=rot, scale=(1, 1, 1.1),
                                     shape='/levels/drivebuild/art/objects/pole_traffic1.dae')
                 scenario.add_object(pole)
             elif obstacle_type is TrafficLightDouble:
                 from math import radians, sin, cos
                 from numpy import dot
                 id_number = randrange(1000)
+                rot = (0, 0, -(obstacle.z_rot + 90))
                 name_light1 = "trafficlight" + str(id_number)
                 name_light2 = "trafficlight" + str(id_number) + 'a'
                 name_pole = "pole" + str(id_number)
-                rad = radians(obstacle.z_rot)
+                rad = radians(rot[2])
```

Comment:
As soon as traffic signs allow to be rotated with xRot and yRot as well, this calculation potentially changes.